### PR TITLE
Fix the bug that the custom GrantedAuthority comparison fails

### PR DESCRIPTION
**Pull request by terminux**
Closes gh-10566

### CODE REVIEW

File: core/src/test/java/org/springframework/security/authorization/AuthorityAuthorizationManagerTests.java [hunk 1]
```diff
@@ -91,6 +93,18 @@ public void hasAnyAuthorityWhenContainNullThenException() {
 				.withMessage("authorities cannot contain null values");
 	}
 
+	@Test
+	public void hasAnyAuthorityWhenUserHasCustomAuthorityThenGrantedDecision() {
```
Comment (marcusdacoregio) [inline core/src/test/java/org/springframework/security/authorization/AuthorityAuthorizationManagerTests.java:97]: Can you also provide a test for when the `Authentication` has the wrong authorities but uses a custom implementation?

File: core/src/test/java/org/springframework/security/authorization/AuthorityReactiveAuthorizationManagerTests.java [hunk 2]
```diff
@@ -88,6 +89,15 @@ public void checkWhenHasAuthorityAndAuthorizedThenReturnTrue() {
 		assertThat(granted).isTrue();
 	}
 
+	@Test
+	public void checkWhenHasCustomAuthorityAndAuthorizedThenReturnTrue() {
```
Comment (marcusdacoregio) [inline core/src/test/java/org/springframework/security/authorization/AuthorityReactiveAuthorizationManagerTests.java:93]: Can you also provide a test for when the `Authentication` has the wrong authorities but uses a custom implementation?

**Review by marcusdacoregio — CHANGES_REQUESTED** [review]
Hi @terminux. Thanks for the PR.

I've left some minimum feedback inline.

File: core/src/test/java/org/springframework/security/authorization/AuthorityAuthorizationManagerTests.java (hunk 1 above)
Comment (terminux) [reply core/src/test/java/org/springframework/security/authorization/AuthorityAuthorizationManagerTests.java]: Done.

File: core/src/test/java/org/springframework/security/authorization/AuthorityReactiveAuthorizationManagerTests.java (hunk 2 above)
Comment (terminux) [reply core/src/test/java/org/springframework/security/authorization/AuthorityReactiveAuthorizationManagerTests.java:93]: Done.

**Review by marcusdacoregio — APPROVED** [review]
Awesome!

**Pull request merged**